client/components/overviewWidget/ImageGallery.jsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import Thumbnails from './Thumbnails.jsx';
+import axios from 'axios';
+
+class ImageGallery extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      productId: null,
+      mainPhotoUrl: '',
+      thumbnails: []
+    };
+  }
+
+  componentDidMount() {}
+
+  componentDidUpdate(prevProps) {
+    if (this.props.product !== prevProps.product) {
+      this.setState({
+        productId: this.props.product.id
```

Comment:
Again making sure this is on purpose as it is discouraged to set state from props by React